tests/llm/clients/test_client_config.py

```
@@ -0,0 +1,266 @@
+        assert not custom.is_closed
+        await custom.aclose()
+
+    @pytest.mark.asyncio
+    async def test_close_closes_owned_client(self):
+        client = OpenAICompatibleClient(base_url="https://api.openai.com/v1", api_key="sk")
+        owned = client._client
+        await client.close()
+        assert owned.is_closed
+
+    def test_invalid_type_raises(self):
+        with pytest.raises(TypeError, match="httpx.AsyncClient"):
```

Ruff (0.15.11)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)